src/Axiom/Set/TotalMap.agda

```diff
@@ -1,34 +1,76 @@
 {-# OPTIONS --safe --no-import-sorts #-}
-{-# OPTIONS -v allTactics:100 #-}
 
-open import Agda.Primitive renaming (Set to Type)
-open import Axiom.Set
+open import Axiom.Set using ( Theory )
 
 module Axiom.Set.TotalMap (th : Theory) where
 
-open import Prelude
+open import Prelude hiding (lookup)
 
-open Theory th
-open import Axiom.Set.Map th
-open import Axiom.Set.Properties th
-open import Axiom.Set.Rel th
-open import Tactic.AnyOf
-open import Tactic.Defaults
+open import Agda.Primitive              using () renaming (Set to Type)
+open import Axiom.Set.Map th            using (left-unique; Map ; mapWithKey-uniq)
+open import Axiom.Set.Rel th            using (Rel ; dom ; dom∈)
+open import Interface.DecEq             using (DecEq ; _≟_)
+open import Relation.Nullary.Decidable  using (yes ; no)
+
+open Theory th    using (_∈_ ; map ; Set ; ∈-map ; ∈-map′)
+open Equivalence  using (to ; from)
 
 private variable A B : Type
 
--- Because of missing macro hygiene, we have to copy&paste this. https://github.com/agda/agda/issues/3819
-private macro
-  ∈⇒P = anyOfⁿᵗ (quote ∈-filter⁻' ∷ quote ∈-∪⁻ ∷ quote ∈-map⁻' ∷ quote ∈-fromList⁻ ∷ [])
-  P⇒∈ = anyOfⁿᵗ (quote ∈-filter⁺' ∷ quote ∈-∪⁺ ∷ quote ∈-map⁺' ∷ quote ∈-fromList⁺ ∷ [])
-  ∈⇔P = anyOfⁿᵗ (quote ∈-filter⁻' ∷ quote ∈-∪⁻ ∷ quote ∈-map⁻' ∷ quote ∈-fromList⁻ ∷ quote ∈-filter⁺' ∷ quote ∈-∪⁺ ∷ quote ∈-map⁺' ∷ quote ∈-fromList⁺ ∷ [])
 
 -- defines a total map for a given set
 total-map : Rel A B → Type
 total-map R = ∀ {a} → a ∈ map proj₁ R
 
-TotalMap : Type → Type → Type
-TotalMap A B = Σ (Rel A B) (λ R → left-unique R × total-map R)
 
-lookupMapᵗ : TotalMap A B → A → B
-lookupMapᵗ (_ , _ , tot) a = proj₂ (proj₁ (∈⇔P (tot {a})))
+record TotalMap (A B : Type) : Type where
+  field  rel : Set (A × B)
+         lun : left-unique rel
+         tot : ∀{a : A} → a ∈ dom rel
+
+
+  MapReduct : Map A B
+  MapReduct = rel , lun
+
+
+  lookup : A → B
+  lookup a = proj₁ ξ
+    where
+    ξ : ∃[ b ] (a , b) ∈ rel
```

Comment:
Please inline this, there's no reason for a `where` block

Reply:
I agree, the `where` block seems unnecessary here.  The reason I left it that way is because when I inline it I need to specify one of the implicit parameters (the `a`, if I remember correctly).  So I guess this is a good example of "broken" inference...?  Anyway, I'll inline it.

Reply:
(I inlined it and somehow it's no longer requiring an implicit parameter... weird.  Must be because the code is quite different now to when I had tried to inline it before.)